src/keystore.cpp

```diff
@@ -93,6 +93,27 @@ bool CBasicKeyStore::AddSpendingKey(const libzcash::SproutSpendingKey &sk)
     return true;
 }
 
+//! Sapling 
+bool CBasicKeyStore::AddSaplingSpendingKey(const libzcash::SaplingSpendingKey &sk)
+{
+    LOCK(cs_SpendingKeyStore);
+    auto fvk = sk.full_viewing_key();
+    
+    // if SaplingFullViewingKey is not in SaplingFullViewingKeyMap, add it
+    if (!AddSaplingFullViewingKey(fvk)){
+        return false;
+    }
+    
+    mapSaplingSpendingKeys[fvk] = sk;
+
+    // Add addr -> SaplingIncomingViewing to SaplingIncomingViewingKeyMap
+    auto ivk = fvk.in_viewing_key();
+    auto addr = sk.default_address();
+    mapSaplingIncomingViewingKeys[addr] = ivk;
```

Comment:
Reminder to self: when we implement ZIP 32, we need to set the `address -> ivk` mapping inside `AddSaplingFullViewingKey()`, not here (which we can't do in this PR because we can't get the default address from `fvk`).